src/services/completions.ts

```diff
@@ -335,6 +328,52 @@ namespace ts.Completions {
         }
     }
 
+    function getJsxCompletionClosingTag(location: Node | undefined, sourceFile: SourceFile) {
+        // This helps us to identify if location is a property access expression,
+        // in this case we wanna make sure if the parent of parent is a jsx closing tag
+        // The following is a possible case:
+        //   var x = <Comp.Item></Comp.Item/*1*/
+        const jsxClosingElement = location && location.parent
+            ? location.parent.parent && isJsxClosingElement(location.parent.parent)
+                ? location.parent.parent
+                : isJsxClosingElement(location.parent)
+                    ? location.parent
+                    : undefined
+            : undefined;
+
+        if (jsxClosingElement) {
+            // In the TypeScript JSX element, if such element is not defined. When users query for completion at closing tag,
+            // instead of simply giving unknown value, the completion will return the tag-name of an associated opening-element.
+            // For example:
+            //     var x = <div> </ /*1*/
+            // The completion list at "1" will contain "div>" with type any
+            // And at `<div> </ /*1*/ >` (with a closing `>`), the completion list will contain "div".
+            // And at property access expressions `<MainComponent.Child> </MainComponent. /*1*/ >` the completion will
+            // return the partial opening tag as in `Child`
+            // And if the property access expression is equal or superceding the opening tag empty string will be provided
+            const hasClosingAngleBracket = !!findChildOfKind(jsxClosingElement, SyntaxKind.GreaterThanToken, sourceFile);
+            const accessExpressionNode = findChildOfKind(jsxClosingElement, SyntaxKind.PropertyAccessExpression, sourceFile);
+            const tagName = jsxClosingElement.parent.openingElement.tagName;
+            let closingTag = tagName.getFullText(sourceFile);

```

Comment:
After thinking a bit more about this. Is it possible that the closing tag includes whitespaces? Do we wanna cover that too?